### PR TITLE
fix(workflow): Default filter state on page load is not working as expected

### PR DESCRIPTION
**Pull request by iProgramStuff**
On initial page load of the alerts listing, the API call to get the alerts wasn't using the default status of "open". However, because the status variable is set afterwards, the UI showed "open" alerts as the selection. This is bad because the results being shown included all status'.

This fixes that by redirecting the user to ?status=open if it's not set.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 1]
```diff
@@ -132,6 +132,13 @@ class IncidentsListContainer extends React.Component<Props> {
    * Incidents list is currently at the organization level, but the link needs to
    * go down to a specific project scope.
    */
+  componentWillUpdate() {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx]: This hook method is deprecated and shouldn't be used. Instead you could do this logic in componentDidUpdate or componentDidMount.

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 2]
```diff
@@ -132,6 +132,13 @@ class IncidentsListContainer extends React.Component<Props> {
...
+    const {params, location, router} = this.props;
+    if (router && location.query.status === undefined) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx]: What about other falsey values like '' and 0?

**Review by billyvg — DISMISSED** [review]
Hmm I think instead of redirecting, I would just change the API request to include a default 
https://github.com/getsentry/sentry/blob/e4535b4d3df4bfb57fe9ab7623d75e06ed2c6168/src/sentry/static/sentry/app/views/alerts/list/index.tsx#L45

**Comment by iProgramStuff** [comment]
> 
> 
> Hmm I think instead of redirecting, I would just change the API request to include a default
> 
I did this at first but the URL didn't change...so I tried the redirect method. I was worried this would have more of a performance impact..and be more confusing to read / understand for the next person.

**Comment by billyvg** [comment]
The render method already treats open as the default so I think it’s fine to have the the api do the same. If we redirect we potentially have the component making two api requests before and after redirect.

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 3]
```diff
@@ -34,15 +34,21 @@ type State = {
   incidentList: Incident[];
 };
 
+function getQueryStatus(status: any) {
+  return status === undefined || status === '' ? DEFAULT_QUERY_STATUS : status;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx]: `status` _could_ be a list if it was included multiple times in the query string. But that might not be a scenario worth worrying about.

**Review by markstory — APPROVED** [review]

**Review by billyvg — APPROVED** [review]
🌮

**Pull request merged**